api/v1alpha1/gateway_types.go

```diff
@@ -179,56 +182,101 @@ type ListenerTLS struct {
+	// LastTransitionTime indicates the last time this condition changed.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
 }
 
 // ListenerStatus is the status associated with each listener block.
 type ListenerStatus struct {
-	// Errors is a list of reasons why a given Listener Spec is
-	// not valid. Errors will be empty if the Spec is valid.
-	Errors []ListenerError `json:"errors"`
-
-	//
-	Address string `json:"address"`
+	// Name is the name of the listener this status refers to.
+	Name string `json:"name"`
+	// Address bound on this listener.
+	Address *ListenerAddress `json:"address"`
+	// Conditions describe the current condition of this listener.
+	Conditions []ListenerCondition `json:"conditions"`
 }
 
-// ListenerErrorReason is the type for errors associated with the
-// listener.
-type ListenerErrorReason string
+// ListenerConditionType is a type of condition associated with the listener.
+type ListenerConditionType string
 
 const (
-	// ErrListenerInvalidSpec is a generic error that is a
-	// catch all for unsupported configurations that do not match a
-	// more specific error. Implementors should try to use more
-	// specific errors instead of this one to give users and
-	// automation a more information.
-	ErrListenerInvalidSpec ListenerErrorReason = "InvalidSpec"
-	// ErrListenerBadAddress indicates the Address
-	ErrListenerBadAddress ListenerErrorReason = "InvalidAddress"
+	// ConditionInvalidListener is a generic condition that is a catch all for
+	// unsupported configurations that do not match a more specific condition.
+	// Implementors should try to use more specific condition instead of this
+	// one to give users and automation a more information.
+	ConditionInvalidListener ListenerConditionType = "InvalidListener"
+	// ConditionListenerNotReady indicates the listener is not ready.
+	ConditionListenerNotReady ListenerConditionType = "ListenerNotReady"
+	// ConditionInvalidAddress indicates the Address is invalid.
+	ConditionInvalidAddress ListenerConditionType = "InvalidAddress"
+	// ConditionNoSuchListener indicates the listener does not exist.
+	ConditionNoSuchListener ListenerConditionType = "NoSuchListener"
 )
 
-// ListenerError is an error status for a given ListenerSpec.
-type ListenerError struct {
-	// Reason is a automation friendly reason code for the error.
-	Reason ListenerErrorReason `json:"reason"`
-	// Message is a human-understandable error message.
+// ListenerCondition is an error status for a given listener.
+type ListenerCondition struct {
+	// Type indicates the type of condition. Can be "InvalidListener",
+	// "InvalidAddress", or "NoSuchListener".
+	Type ListenerConditionType `json:"type"`
+	// Status describes the current state of this condition. Can be "True",
+	// "False", or "Unknown".
+	Status core.ConditionStatus `json:"status"`
+	// Message is a human-understandable message describing the condition.
 	Message string `json:"message"`
```

Comment:
It was, but that was only because I'd seen an example of conditions used that way. It seems like message is optional most of the time though. Updated to make it optional here as well.